sql/parser/eval.go

```diff
@@ -134,15 +137,23 @@ var binOps = map[binArgs]binOp{
 		},
 	},
 	binArgs{Plus, dateType, intervalType}: {
-		returnType: DummyTimestamp,
+		returnType: DummyDate,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			days, err := inDays(right.(DInterval).Duration)
+			if err != nil {
+				return DummyDate, err
+			}
+			return DDate(left.(DDate) + days), nil
```

Comment:
if you want the postgres behavior, you have to explicitly cast the date to a timestamp and then add an interval. We basically only allow dates to be modified in days specified as in interval. Another option is to forget about the whole interval crap and use an integer for days which is what postgres does.